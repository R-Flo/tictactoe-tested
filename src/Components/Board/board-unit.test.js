import React from 'react'
import Board from './board'
import {shallow, mount} from 'enzyme'
import '../../setUpTests';

jest.mock('../../Components/Square/square', () => ()=> <div className="square">X</div>);


describe('square', () => {

  let squares = Array(9).fill(null);

  it('renders without crashing', () => {

    const wrapper = shallow(<Board squares={squares}/>);

    expect(wrapper).toMatchSnapshot();

  });

  it('Shows all squares', () => {
    const wrapper = mount(<Board squares={squares}/>);
    expect(wrapper.find('.square').length).toEqual(9);
  });


});


