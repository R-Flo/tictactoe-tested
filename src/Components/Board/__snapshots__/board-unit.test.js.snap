// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    squares={
        Array [
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="board-row"
>
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
</div>,
        <div
          className="board-row"
>
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
</div>,
        <div
          className="board-row"
>
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
          <Unknown
                    onClick={[Function]}
                    value={null}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              onClick={[Function]}
              value={null}
/>,
            <Unknown
              onClick={[Function]}
              value={null}
/>,
            <Unknown
              onClick={[Function]}
              value={null}
/>,
          ],
          "className": "board-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              onClick={[Function]}
              value={null}
/>,
            <Unknown
              onClick={[Function]}
              value={null}
/>,
            <Unknown
              onClick={[Function]}
              value={null}
/>,
          ],
          "className": "board-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              onClick={[Function]}
              value={null}
/>,
            <Unknown
              onClick={[Function]}
              value={null}
/>,
            <Unknown
              onClick={[Function]}
              value={null}
/>,
          ],
          "className": "board-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="board-row"
>
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
</div>,
          <div
            className="board-row"
>
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
</div>,
          <div
            className="board-row"
>
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
            <Unknown
                        onClick={[Function]}
                        value={null}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                onClick={[Function]}
                value={null}
/>,
              <Unknown
                onClick={[Function]}
                value={null}
/>,
              <Unknown
                onClick={[Function]}
                value={null}
/>,
            ],
            "className": "board-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                onClick={[Function]}
                value={null}
/>,
              <Unknown
                onClick={[Function]}
                value={null}
/>,
              <Unknown
                onClick={[Function]}
                value={null}
/>,
            ],
            "className": "board-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                onClick={[Function]}
                value={null}
/>,
              <Unknown
                onClick={[Function]}
                value={null}
/>,
              <Unknown
                onClick={[Function]}
                value={null}
/>,
            ],
            "className": "board-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
